tests/difference/core/quint_model/ccv.qnt

```diff
@@ -0,0 +1,619 @@
+        RegisterNewOutstandingPackets(newOutstandingPackets),
+        val packet = outstandingPacketsToConsumer.get(consumer).head()
+        all {
+            // update the running validator set, but not the history yet,
+            // as that only happens when the next block is started
+            runningValidatorSet' = runningValidatorSet.set(consumer, packet.validatorSet),
+            // add the new packet and store its maturation time
+            val newMaturationTimes = maturationTimes.get(consumer).put(packet, curChainTimes.get(consumer) + UnbondingPeriod.get(consumer))
+            maturationTimes' = maturationTimes.set(consumer, newMaturationTimes)
+        },
+        PROVIDER_NOOP,
+        votingPowerHistories' = votingPowerHistories,
+        outstandingPacketsToProvider' = outstandingPacketsToProvider,
+        // consumer statusses do not change
+        consumerStatus' = consumerStatus,
+        // chain times do not change
+        curChainTimes' = curChainTimes,
+        // the validator set was not changed by this action (but might have been changed before in this block)
+        providerValidatorSetChangedInThisBlock' = providerValidatorSetChangedInThisBlock
+    }
+
+    // ends the current block and starts the next block for a given chain.
+    action endAndBeginBlock(chain: Chain): bool = any {
+        all {
+          chain == ProviderChain,
+          endAndBeginBlockForProvider,
+        },
+        all {
+            chain != ProviderChain,
+            endAndBeginBlockForConsumer(chain),
+        }
+    }
+
+    // gets the updated history for the current chain when ending a block, i.e. the
+    // running validator set is added to the history if different from the last one.
+    def getUpdatedHistory(chain: Chain): List[ValidatorSet] =
+        // update voting power history if the validator set changed
+        val newValidatorSet = runningValidatorSet.get(ProviderChain)
+        val oldValidatorSet = votingPowerHistories.get(ProviderChain).head()
+        if (newValidatorSet != oldValidatorSet) 
+            votingPowerHistories.get(ProviderChain).prepend(newValidatorSet)
+        else
+            votingPowerHistories.get(ProviderChain)
+
+
+    action endAndBeginBlockForProvider(): bool = all {
+        // update the voting power history
+        votingPowerHistories' = votingPowerHistories.set(ProviderChain, getUpdatedHistory(ProviderChain)),
+        // the running validator set is now for sure the current validator set,
+        // so start with it in the next block
+        runningValidatorSet' = runningValidatorSet,
+        // send VSCPackets to consumers
+        val newOutstandingPackets = 
+            // if running validator set is considered to have changed
+            if (providerValidatorSetChangedInThisBlock)
+                // then send a packet to each running consumer
+                outstandingPacketsToConsumer.keys().mapBy(
```

Comment:
What if it's a new consumer chain without any outstanding packets? That would mean that `outstandingPacketsToConsumer` has no key with that consumer, right? 
Why not use `ConsumerChains.mapBy`?